### DISCUSSION TITLE
How to create a separate transparent window on click?

### DISCUSSION THREAD

**Discussion by HuakunShen** [tauri-apps/tauri#7130] [Q&A] [answered]
I want to create a transparent window with no title bar and no background in order to show an animation.

I could set the main window to transparent and no decoration, but when a new window is created, it always has a background and title bar.

The code is like this

```ts
import { WebviewWindow } from "@tauri-apps/api/window";
...
const win = new WebviewWindow("win-animate", {
      url: "/animation",
    });
await win.setDecorations(false);
```

## Screenshot

Here is a screenshot of the result. On the left, it's the transparent main window; on the right, it's the newly created window which always has background and title bar.

The green rectangle is where the main window is, the red rectangle indicate the button to click to create the new window.

![image](https://github.com/tauri-apps/tauri/assets/33727687/51e867e5-21af-472d-847b-6080aed63e9b)

Thank you.

**Comment by FabianLars** [comment]
I feel like this is a timin issue. `new WebviewWindow` is techincally async so it may not be done when you call win.setDecorations(false) 🤔 
Does setting it in the constructor via `decorations: false` not work for you?

**Reply by HuakunShen** [reply]
Actually, `win.setDecorations(false)` worked for me. But there is not a `setTransparent` API.

**Reply by HuakunShen** [reply, marked as answer]
Oh, it works if I use the constructor way.

```ts
const window = new WebviewWindow("win-animate", {
      url: "/animation",
      transparent: true,
      decorations: false
    });
```

Thank you! @FabianLars
